components/automate-ui/src/app/page-components/services-sidebar/services-sidebar.component.ts

```diff
@@ -117,4 +119,20 @@ export class ServicesSidebarComponent implements OnInit, OnDestroy {
     }
   }
 
+  public toggleMoreErrorMsg(index: number) {
+    if (this.activeErrorAccordions.includes(index)) {
+      this.activeErrorAccordions = this.activeErrorAccordions.filter(n => n !== index);
+    } else {
+      this.activeErrorAccordions.push(index);
+    }
+  }
+
+  public toggleMoreHealthMsg(index: number) {
+    if (this.activeHealthAccordions.includes(index)) {
+      this.activeHealthAccordions = this.activeHealthAccordions.filter(n => n !== index);
+    } else {
+      this.activeHealthAccordions.push(index);
+    }
+  }
+
```

Comment:
I wanted a way to expand/collapse accordions and to be able to have multiple accordions open/collapsed at different times.  I felt like the best way to accomplish this was to store the index of active accordions in a public array, and then have the template check for whether or not the array contained that particular index and attach an `active` class accordingly.  This way, the CSS handles the actual height of the active accordion.

Reply:
I should add that although these functions accomplish the exact same task - I felt it was easier to follow/maintain by keeping them separate and not having to pass in Arrays or follow any kind of extended logic.  Open to suggestions though.

Reply:
good idea - I chose to use close/open accordion for a bit more specificity of whats goin on in the template.